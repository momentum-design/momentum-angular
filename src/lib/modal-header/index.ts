export * from './modal-header.module';
export * from './modal-header.component';
