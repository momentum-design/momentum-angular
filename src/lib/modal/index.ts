export * from './modal.module';
export * from './modal.component';
export * from './modal-body.module';
export * from './modal-body.component';
export * from './modal-footer.module';
export * from './modal-footer.component';
export * from './modal-header.module';
export * from './modal-header.component';

