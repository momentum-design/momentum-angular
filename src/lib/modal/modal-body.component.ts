/** @component modal-body */

import { Component, Input, HostBinding } from '@angular/core';

@Component({
  selector: 'md-modal-body',
  template: `
    <ng-content></ng-content>
  `,
  styles: [],
})
export class ModalBodyComponent {
  /** @prop css class names | '' | '' */
  @Input() public class: string = '';

  @HostBinding('class') get className(): string {
    return 'md-modal__body' + `${(this.class && ` ${this.class}`) || ''}` + ``;
  }
}
