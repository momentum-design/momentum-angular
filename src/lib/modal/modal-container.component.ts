/** @component modal-container */

import {
  Component,
  OnInit
} from '@angular/core';
import { ModalContent, ModalRef } from './modal-ref';

@Component({
  selector: 'md-modal-container',
  template: `
  <ng-container>
    <ng-container *ngComponentOutlet="content"></ng-container>
  </ng-container>
  `,
})
export class ModalContainerComponent  implements OnInit {

   public content: ModalContent;
   public context;

   constructor( private modalRef: ModalRef) {}

  ngOnInit() {
    this.content = this.modalRef.content;
    this.context = {
      close: this.modalRef.close.bind(this.modalRef)
    };

  }

}
