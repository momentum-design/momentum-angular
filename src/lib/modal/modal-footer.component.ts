/** @component modal-footer */

import { Component,  Input, HostBinding } from '@angular/core';

@Component({
  selector: 'md-modal-footer',
  template: `
    <ng-content></ng-content>
  `,
  styles: [],
})
export class ModalFooterComponent  {
  /** @prop css class names | '' | '' */
  @Input() public class: string = '';

  @HostBinding('class') get className(): string {
    return (
      'md-modal__footer' + `${(this.class && ` ${this.class}`) || ''}` + ``
    );
  }
}
