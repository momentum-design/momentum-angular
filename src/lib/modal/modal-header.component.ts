/** @component modal-header */

import {
  Component,
  Input,
  HostBinding,
  HostListener,
} from '@angular/core';
import { ModalRef } from './modal-ref';


@Component({
  selector: 'md-modal-header',
  template: `
  <span class="md-modal__title">{{headerLabel}}</span>
  <span *ngIf="message" class="md-modal__message">{{
    message
  }}</span>
  <button
    *ngIf="showCloseButton"
    class="md-close md-modal__close"
    (click)="close()"
    (keydown)="handleKeydown($event)"
    aria-label="close"
  >
  </button>
  `,
})
export class ModalHeaderComponent {
  /** @prop Optional css class string | '' */
  @Input() public class: string = '';
  /** @prop ModalHeader label text | '' */
  @Input() public headerLabel: string = '';
  /** @prop  Modal message | '' */
  @Input() public message: string = '';
  /** @prop show/hide close button | true */
  @Input() public showCloseButton: boolean = true;

  @HostBinding('class') get className(): string {
    return (
      'md-modal__header' + `${(this.class && ` ${this.class}`) || ''}` + ``
    );
  }


  @HostListener('document:keydown.escape', ['$event']) handleKeydown(
    _: KeyboardEvent
  ) {
    this.close();
  }

  constructor(private modalRef: ModalRef) {}

  close() {
    this.modalRef.close();
  }
}
