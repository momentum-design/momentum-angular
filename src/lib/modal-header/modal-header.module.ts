import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ModalHeaderComponent } from './modal-header.component';

@NgModule({
  declarations: [ModalHeaderComponent],
  imports: [CommonModule],
  exports: [ModalHeaderComponent],
})
export class ModalHeaderModule {}
