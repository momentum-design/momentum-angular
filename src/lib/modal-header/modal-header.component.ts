import {
  Component,
  OnInit,
  Input,
  OnDestroy,
  HostBinding,
} from '@angular/core';
import { ModalService } from '../modal/modal.service';
import { Subscription } from 'rxjs';

@Component({
  selector: 'cui-modal-header',
  template: `
    <ng-content></ng-content>
    <span *ngIf="hideNative" class="cui-modal__title">{{ headerLabel }}</span>
    <span *ngIf="hideNative && message" class="cui-modal__message">{{
      message
    }}</span>
    <button
      *ngIf="showCloseButton"
      cui-button
      aria-label="Close Modal"
      (click)="closeModal()"
      class="cui-close cui-modal__close"
    ></button>
  `,
  styles: [],
  providers: [],
})
export class ModalHeaderComponent implements OnInit, OnDestroy {
  /** @option Optional css class string | '' */
  @Input() public class: string = '';
  /** ModalHeader label text | '' */
  @Input() public headerLabel: string = '';
  /** Modal message | '' */
  @Input() public message: string = '';
  /** show/hide close button | true */
  @Input() public showCloseButton: boolean = true;

  @HostBinding('class') get className(): string {
    return (
      'cui-modal__header' + `${(this.class && ` ${this.class}`) || ''}` + ``
    );
  }

  private subscription: Subscription;
  public hideNative: boolean = true;

  constructor(private modalService: ModalService) {}

  ngOnInit() {
    this.hideNative = this.headerLabel !== '';
  }

  ngOnDestroy() {
    // prevent memory leak when component destroyed
    if (this.subscription) {
      this.subscription.unsubscribe();
    }
  }

  closeModal = () => {
    this.modalService.setModalStatus(false);
  }
}

/**
 * @component modal-header
 * @section ng-content
 * @angular
 *
<cui-modal>
  <cui-modal-header headerLabel='large'>
    <a>My Header</a>
  </cui-modal-header>
</cui-modal>
 */

/**
 * @component modal-header
 * @section headerLabel
 * @angular
 *
<cui-modal>
  <cui-modal-header headerLabel='large'></cui-modal-header>
</cui-modal>
 */

/**
 * @component modal-header
 * @section message
 * @angular
 *
<cui-modal>
  <cui-modal-header headerLabel='large' message='need headerLabal'></cui-modal-header>
</cui-modal>
 */

/**
 * @component modal-header
 * @section class
 * @angular
 *
<cui-modal>
  <cui-modal-header class='myClass'></cui-modal-header>
</cui-modal>
 */

/**
 * @component modal-header
 * @section showCloseButton
 * @angular
 *
<cui-modal>
  <cui-modal-header [showCloseButton]='false'></cui-modal-header>
</cui-modal>
 */
