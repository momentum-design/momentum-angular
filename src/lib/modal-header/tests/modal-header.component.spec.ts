import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ModalService } from '../../modal/modal.service';
import { ModalHeaderComponent } from '../modal-header.component';

describe('ModalHeaderComponent', () => {
  let component: ModalHeaderComponent;
  let fixture: ComponentFixture<ModalHeaderComponent>;
  let testElement: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ModalHeaderComponent],
      imports: [],
      providers: [ModalService],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ModalHeaderComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create and match snapshot', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render classname', () => {
    component.class = 'textAddOn';
    fixture.detectChanges();
    testElement = fixture.nativeElement;
    expect(testElement.className).toContain('textAddOn');
  });

  it('should render headerLabel', () => {
    component.headerLabel = 'headerLabel';
    component.ngOnInit();
    fixture.detectChanges();
    testElement = fixture.nativeElement.getElementsByTagName('span')[0];
    expect(testElement.innerHTML).toContain('headerLabel');
  });

  it('should render message', () => {
    component.headerLabel = 'headerLabel';
    component.message = 'message';
    component.ngOnInit();
    fixture.detectChanges();
    testElement = fixture.nativeElement.getElementsByTagName('span')[1];
    expect(testElement.innerHTML).toContain('message');
  });
});
