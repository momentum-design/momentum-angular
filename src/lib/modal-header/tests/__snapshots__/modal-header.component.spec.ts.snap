// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeaderComponent should create and match snapshot 1`] = `
<cui-modal-header
  class=""
  closeModal={[Function Function]}
  headerLabel=""
  hideNative="false"
  message=""
  modalService={[Function ModalService]}
  showCloseButton={[Function Boolean]}
>
  <button
    aria-label="Close Modal"
    class="cui-close cui-modal__close"
    cui-button=""
  />
</cui-modal-header>
`;
